### PR TITLE
Add payload structure to readme

### PR DESCRIPTION
**Pull request by Qmpzlawasd**

### CODE REVIEW

**Review by AdelaCorbeanu — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,26 +1,45 @@
...
+### IPC protocol from client to daemon
+
 - 2 bytes - `"da"` (hex: `64 61`) magic bytes
 - 1 byte - command
 - 8 bytes - payload length
 - `<payload_length>` bytes - payload
+
+### Payload Structure
+
+| COMMAND     | PAYLOAD STRUCTURE                                             |
```
Comment (tezkerek) [inline README.md]: Initcap these

File: README.md [hunk 2]
```diff
@@ -1,26 +1,45 @@
...
 
-### IPC protocol
+### IPC protocol from client to daemon
+
 - 2 bytes - `"da"` (hex: `64 61`) magic bytes
 - 1 byte - command
 - 8 bytes - payload length
 - `<payload_length>` bytes - payload
+
+### Payload Structure
```
Comment (tezkerek) [inline README.md]: Make it a subsection of IPC protocol (`##### Payload Structure`)

File: README.md [hunk 3]
```diff
@@ -1,26 +1,45 @@
...
+  - (2): `pthread_cancel`? or running state + polling?
+  - (3): `pthread_suspend`? or mutexes?
+  - (4): query state
+  - task progress - query number of files in first level? all levels?
+  - traversal
+    - nftw
+    - don't follow symlinks
 - [ ] printing results in client
-    - pretty printing can be complex
+  - pretty printing can be complex
```
Comment (tezkerek) [inline README.md]: Keep the 4 space indent.

File: README.md [hunk 4]
```diff
@@ -1,26 +1,45 @@
...
+
+| COMMAND     | PAYLOAD STRUCTURE                                             |
+| ----------- | ------------------------------------------------------------- |
+| CMD_ADD     | `1` byte - priority <br/> `<payload_length> - 1` bytes - path |
+| CMD_SUSPEND | `8` bytes - id                                                |
+| CMD_RESUME  | `8` bytes - id                                                |
+| CMD_REMOVE  | `8` bytes - id                                                |
+| CMD_INFO    | `8` bytes - id                                                |
+| CMD_LIST    | `0` bytes - id                                                |
+| CMD_PRINT   | `8` bytes - id                                                |
```
Comment (tezkerek) [inline README.md]: Maybe backtick the `CMD_*`s.

**Review by tezkerek — CHANGES_REQUESTED** [review]

**Review by tezkerek — APPROVED** [review]

**Pull request merged**